File: update-tools.py
import subprocess
import platform

def run_command(command):
    try:
        # Use shell=True on Windows for compatibility
        result = subprocess.run(
            command, 
            check=True, 
            stdout=subprocess.PIPE, 
            stderr=subprocess.PIPE, 
            text=True, 
            shell=(platform.system() == "Windows")
        )
        print(result.stdout)
    except subprocess.CalledProcessError as e:
        print(f"Error executing {' '.join(command)}:\n{e.stderr}")

def print_line():
    print("--------------------------------------------------")

def do_work(command, name):
    print_line()
    print(f"Running: {command}")
    print(f"Updating {name}...")
    print("Before update: ")
    run_command([name, "--version"])
    print("updating...")
    run_command(command)
    print("After update: ")
    run_command([name, "--version"])
    print_line()

# Update Bun
do_work(["bun", "--version"], "Bun")

# Update pnpm
do_work(["pnpm", "--version"], "pnpm")

# Update pip
do_work(["pip", "--version"], "pip")

# Update Deno
do_work(["deno", "--version"], "Deno")

File: update-tools.ts
// // update-tools.ts

// const exec = async (cmd: string[]) => {
//     const process = Deno.run({
//         cmd,
//         stdout: "piped",
//         stderr: "piped",
//     });

//     const { code } = await process.status();
//     const rawOutput = await process.output();
//     const rawError = await process.stderrOutput();

//     if (code === 0) {
//         console.log(new TextDecoder().decode(rawOutput));
//     } else {
//         console.error(`Error executing ${cmd.join(" ")}`);
//         console.error(new TextDecoder().decode(rawError));
//     }

//     process.close();
// };

// // Update Bun
// await exec(["bun", "upgrade"]);

// // Update Node and npm (assumes Node and npm are installed via nvm)
// await exec(["nvm", "install", "node", "--reinstall-packages-from=node"]);
// await exec(["npm", "install", "-g", "npm"]);

// // Update pnpm
// await exec(["pnpm", "install", "-g", "pnpm"]);

// // Update Python and pip
// await exec(["pyenv", "install", "--update"]);  // If using pyenv for Python version management
// await exec(["pip", "install", "--upgrade", "pip"]);

// // Update Deno
// await exec(["deno", "upgrade"]);

// // Export to mark this as a module
// export {};
